feat(docs): Add clarification of trace completeness with rate limiting (#10299)
getsentry/sentry-docs: docs/concepts/key-terms/tracing/distributed-tracing.mdx

Put simply: as a result of this head-based approach, where the decision is made once in the originating service and passed to all subsequent services, either all of the transactions for a given trace are collected, or none are, so there shouldn't be any incomplete traces.

<Alert level="warning">

> Dynamic sampling ensures complete traces by retaining all transactions associated with a trace if the head transaction is preserved. Despite dynamic sampling providing trace completeness, transactions or other items (errors, replays, etc.) may still be missing from a trace when rate limiting or quota enforcement drops one or more transactions. Rate limiting drops items without regard for the trace, making each decision independently and potentially resulting in broken traces.

Reviewer: ```suggestion
Dynamic sampling ensures complete traces by retaining all transactions associated with a trace if the head transaction is preserved. Despite dynamic sampling providing trace completeness, transactions or other items (errors, replays, etc.) may still be missing from a trace when rate limiting or quota enforcement drops them. Rate limiting drops items without regard for the trace, making each decision independently and potentially resulting in broken traces.
```
